0017-negotiation-and-execution-protocol.adoc

```diff
@@ -0,0 +1,544 @@
+This is meant for use cases, where the maker did not suggest expiries in the offer, but the taker does not want to fallback to the default expiries of the execution.
+
+The execution protocol may be specified by the taker if the maker specified more than one in his offer.
+
+==== Shortcomings
+
+This first simple negotiation protocol does not really specify an orderbook in the sense that a user can compare ALL available offers.
+A taker can only process the offers of known makers.
+Every taker is responsible for keeping its own list of known makers and evaluate the offers.
+There may exist unknown makers that would offer a better rate.
+
+=== Changes to the existing execution protocol RFC002/RFC003
+
+Constraints:
+
+. Taker defaults to Alice, Maker defaults to Bob
+. The taker has received connection information to contact the maker for execution in the negotiation phase.
+
+==== Input parameters
+
+===== Execution Parameter Definition
+
+For the execution phase as specified by RFC003 we need the parameters for creating the HTLCs on both sides:
+
+* alpha-ledger
+* alpha-asset
+* alpha-amount
+* beta-ledger
+* beta-asset
+* beta-amount
+* expiries
+* hash(secret)
+* identities
+
+In order to come up with these parameters we need:
+
+* role (to know who comes up with the secret)
+
+In order to enable more protocols in the future we need:
+
+* protocol
+
+===== Negotiation Output
+
+From the negotiation phase we have the following output parameters:
+
+[%header,cols=1*]
+|===
+|required
+
+|offer id
+|valid until
+|bid-ledger
+|bid-asset
+|bid-amount
+|ask-ledger
+|ask-asset
+|ask-amount
+|_execution_ protocol
+|_execution_ connection-info (know only to taker)
+|_execution_ role
+|===
+
+[%header,cols=1*]
+|===
+|optional
+
+|_execution_ expiries
+|===
+
+The taker has received the execution connection information from the maker during the negotiation phase. Thus, the taker has to make the first move.
+
+===== Parameter Mapping
+
+This section defines the input parameters for cnd and
+
+[%header,cols=2*]
+|===
+|required (cnd)
+|negotiation
+
+|offer id
+|offer id
+
+|beta-ledger
+|bid-ledger
+
+|beta-asset
+|bid-asset
+
+|beta-amount
+|bid-amount
+
+|alpha-ledger
+|ask-ledger
+
+|alpha-asset
+|ask-asset
+
+|alpha-amount
+|ask-amount
+
+|protocol
+|_execution_ protocol
+
+| connection-info
+|_execution_ connection-info
+
+|role
+|_execution_ role
+
+|===
+
+[%header,cols=2*]
+|===
+|optional (cnd)
+|negotiation
+
+|expiries
+|_execution_ expiries
+
+|identities
+|
+|===
+
+Constraints:
+
+* The connection-info is only provided to the cnd in the role of Alice.
+* If the expiries were not part of the negotiation output cnd will come up with defaults.
+* Identities might be provided by the user. If no identities are provided the cnd will come up with defaults.
+
+==== Communication changes
+
+Instead of a request-response model we propose a messaging model where both parties supply the same parameter to the execution on both sides.
```

Comment:
Please clarify what you mean with "supply".
Do you mean the Maker App supply to the Maker cnd the same parameters than the Taker App supplies to the Taker cnd?